python/run-tests.py

```diff
@@ -228,6 +228,10 @@ def get_default_python_executables():
 def split_and_validate_testnames(testnames):
     testnames_to_test = []
 
+    py4j_module_path = os.path.join(SPARK_HOME, "python/lib/py4j-0.10.9.9-src.zip")
```

Comment:
Hm, I believe we're already handling this via `bin/pyspark` script (see `run_individual_python_test`).

Reply:
Does it fail when it imports? if the case, `importlib.util.find_spec(name) is not None` should work

Reply:
Yes, we did this in `bin/pyspark` so we need it here. `importlib.util.find_spec(name)` fails because it can't find `py4j`. It can find `pyspark` that's why the errors.

Basically when the script tries to locate the test module. It can find `pyspark`, but not `pyspark.sql` (because `py4j` is not there). Then it believes the test should be split as `pyspark sql.xxx`. We need `py4j` in this specific script (that runs before `bin/pyspark`) to check the test module properly.